cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -190,11 +191,12 @@ VAR_LEN_TYPES(IS_NOT_NULL, isnotnull)
 #undef IS_NOT_NULL
 
 FORCE_INLINE
-char* substr_utf8_int64_int64(int64 context, const char* input, int32 in_len,
-                              int64 offset64, int64 length, int32* out_len) {
+const char* substr_utf8_int64_int64(int64 context, const char* input, int32 in_len,
+                                    int64 offset64, int64 length, int32* out_len) {
   if (length <= 0 || input == nullptr || in_len <= 0) {
     *out_len = 0;
-    return nullptr;
+    const char* empty_str = "";
+    return const_cast<char*>(empty_str);
```

Comment:
do you need this cast anymore now that the function returns `const char*`?

Reply:
changed. thanks for pointing out